### DISCUSSION TITLE
Error type: URL is not defined

### DISCUSSION THREAD

**Discussion by laraamadeo** [zheeeng/export-from-json#226] [General]
Hi, I'm doing a custom plugin for figma so I don't have space for debugging. I have a color variables json that want to export each time the plugin is run. 
My code is:
```
    const fileName = `json_colors_${platform}`
    const exportType = exportFromJSON.types.json
    exportFromJSON({ json, fileName, exportType })
```

This error is showing: 

<img width="523" alt="image" src="https://github.com/zheeeng/export-from-json/assets/126255862/16a778f3-bdd9-4602-bdd2-151e3238d08e">

Thanks!

**Comment by zheeeng** [comment]
The error occurs in calling download file function. And it seems that Figma plugin runs in the service worker which can't access BOM API directly. You can pass a custom processor which use Figma Plugin API to achieve downloading instead of using the default BOM downloader.
